### ISSUE TITLE
Handle missing authentication

### ISSUE THREAD

**Issue by christian-heusel** [FachschaftMathPhysInfo/altklausur-ausleihe#26]
Currently the application just doesnt work if no (valid) JWT token is present, instead one should be redirected to the URL where a JWT token can be obtained.

**Comment by christian-heusel** [comment]
@henrixapp is such an issue normally resolved in front or backend?

The two solutions would be:
1. Frontend:
    1. Check on backend endpoint whether the current JWT token is still valid
    2. If not -> Redirect to moodle page  
2. Backend: 
    1. For every invalid invalid token, try to redirect the frontend to the moodle page. Question here is, wheter this is possible / sane :D

**Comment by henrixapp** [comment]
I would not redirect users automatically to the Moodle course. Instead the frontend should show a message explaining the situation and offer a link to Moodle. The backend should only respond to request with valid status codes (403 for forbidden, I think) and not by redirecting.

**Comment by TomTomRixRix** [comment]
Okay, we can display a dialog in the frontend explaining the current situation. Is there a specific response from the backend that can be handled by the front end accordingly? Even with different error types and messages? Or do we simply display a general information dialog with a button to the moodle page?

**Comment by christian-heusel** [comment]
Yes, the backendcurrently sets [HTTP Unauthorized 401](https://developer.mozilla.org/de/docs/Web/HTTP/Status/401) and answers "no token found"  / "token is unauthorized" in plaintext. (see the cURL example below)

If this behaviour needs to change I can implement it as you want 😊 

```
curl -iX POST  -H "Cookie: jwt=SOME.JWT.TOKEN" https://altklausuren.mathphys.info/query
HTTP/1.1 401 Unauthorized
Date: Wed, 29 Sep 2021 22:47:26 GMT
Server: Apache
Content-Type: text/plain; charset=utf-8
Vary: Origin
X-Content-Type-Options: nosniff
Content-Length: 22

token is unauthorized
```
```
curl -iX POST https://altklausuren.mathphys.info/query             
HTTP/1.1 401 Unauthorized
Date: Wed, 29 Sep 2021 22:51:05 GMT
Server: Apache
Content-Type: text/plain; charset=utf-8
Vary: Origin
X-Content-Type-Options: nosniff
Content-Length: 15

no token found
```
